partiql-value/src/lib.rs

```diff
@@ -23,13 +26,158 @@ pub enum Value {
     Boolean(bool),
     Integer(i64),
     Real(OrderedFloat<f64>),
+    Decimal(RustDecimal),
     String(Box<String>),
     Blob(Box<Vec<u8>>),
     List(Box<List>),
     Bag(Box<Bag>),
     Tuple(Box<Tuple>),
 }
 
+impl ops::Add for Value {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self::Output {
```

Comment:
For arithmetic ops, if there are different operand types, coerces the operand in the following way int -> real (if no decimal among operands) -> decimal (if there's a decimal among operands).